### PR TITLE
Added Generic map handler

### PR DESCRIPTION
**Pull request by lxgr-linux**
This is a feature that may be used later to generate playmaps directly from strings.
Therefore I needed to reorganize a lot of stuff.

Please test in depth.

### CODE REVIEW

File: pokete_classes/npcs.py [hunk 1]
```diff
@@ -213,13 +209,13 @@ def action(self):
...
             time.sleep(0.7)
             self.exclamate()
             self.walk_point(self.fig.x, self.fig.y)
             if any(poke.hp > 0 for poke in self.fig.pokes[:6]):
                 self.text(self.texts)
-                winner = self.fight([poke for poke in self.fig.pokes[:6]
+                winner = fm.fight([poke for poke in self.fig.pokes[:6]
                                     if poke.hp > 0][0],
                                     self.poke,
                                     info={"type": "duel", "player": self})
```
Comment (MaFeLP) [inline pokete_classes/npcs.py:221]: Due to the change in line 218, these lines need to be unindented by two spaces:

```suggestion
                                   if poke.hp > 0][0],
                                  self.poke,
                                  info={"type": "duel", "player": self})
```

File: pokete_classes/ob_maps.py [hunk 2]
```diff
@@ -0,0 +1 @@
+ob_maps = None
```
Comment (MaFeLP) [inline pokete_classes/ob_maps.py:1]:
```suggestion
ob_maps = None

if __name__ == "__main__":
    print("\033[31;1mDo not execute this!\033[0m")
```

File: pokete.py [hunk 3]
```diff
@@ -416,8 +279,8 @@ def __init__(self, _si):
         self.caught_pokes = _si.get("caught_poketes", [])
         self.visited_maps = _si.get("visited_maps", ["playmap_1"])
         self.used_npcs = _si.get("used_npcs", [])
-        self.last_center_map = ob_maps[_si.get("last_center_map", "playmap_1")]
-        self.oldmap = ob_maps[_si.get("oldmap", "playmap_1")]
+        self.last_center_map = obmp.ob_maps[_si.get("last_center_map", "playmap_1")]
```
Comment (MaFeLP) [inline pokete.py:282]: line too long

```suggestion
        self.last_center_map = obmp.ob_maps[_si.get("last_center_map",
                                                    "playmap_1")]
```

File: pokete.py [hunk 4]
```diff
@@ -734,7 +425,7 @@ def __call__(self):
 {obj.attack_dict['name']}'?"):
                                     ex_cond = True
                                     while ex_cond:
-                                        index = deck(6, label="Your deck",
+                                        index = deck.deck(6, label="Your deck",
                                                      in_fight=True)
```
Comment (MaFeLP) [inline pokete.py:429]: Due to adding `deck.` in line 428, this line need to be indented by further five spaces.

```suggestion
                                                          in_fight=True)
```

File: pokete.py [hunk 5]
```diff
@@ -1246,48 +905,49 @@ def game(_map):
...
                     exiter()
             elif _ev.get() == "':'":
                 _ev.clear()
-                inp = text_input(movemap.code_label, movemap, ":",
-                                 movemap.width,
-                                 (movemap.width - 2) * movemap.height - 1)[1:]
-                movemap.code_label.outp(figure.map.pretty_name)
+                inp = text_input(mvp.movemap.code_label, mvp.movemap, ":",
+                                 mvp.movemap.width,
+                                 (mvp.movemap.width - 2) * mvp.movemap.height - 1)[1:]
```
Comment (MaFeLP) [inline pokete.py:918]: Line too long:

```suggestion
                                 (mvp.movemap.width - 2)
                                 * mvp.movemap.height - 1)[1:]
```

File: pokete.py [hunk 6]
```diff
@@ -1246,48 +905,49 @@ def game(_map):
...
         time.sleep(0.05)
-        for statement, x, y in zip([figure.x + 6 > movemap.x + movemap.width,
-                                    figure.x < movemap.x + 6,
-                                    figure.y + 6 > movemap.y + movemap.height,
-                                    figure.y < movemap.y + 6],
+        for statement, x, y in zip([figure.x + 6 > mvp.movemap.x + mvp.movemap.width,
+                                    figure.x < mvp.movemap.x + 6,
+                                    figure.y + 6 > mvp.movemap.y + mvp.movemap.height,
+                                    figure.y < mvp.movemap.y + 6],
                                    [1, -1, 0, 0], [0, 0, 1, -1]):
```
Comment (MaFeLP) [inline pokete.py:930]: Lines too long

```suggestion
        for statement, x, y in zip([figure.x + 6 > mvp.movemap.x
                                    + mvp.movemap.width,
                                    figure.x < mvp.movemap.x + 6,
                                    figure.y + 6 > mvp.movemap.y
                                    + mvp.movemap.height,
                                    figure.y < mvp.movemap.y + 6],
```

File: pokete.py [hunk 7]
```diff
@@ -1246,48 +905,49 @@ def game(_map):
...
+        figure.name = ask_text(mvp.movemap,
                                "Welcome to Pokete!\nPlease choose your name!\n",
                                "Name:", "", "Name", 17)
-    movemap.name_label_rechar(figure.name)
-    movemap.text(4, 3, [" < Hello my child.",
+    mvp.movemap.name_label_rechar(figure.name)
+    mvp.movemap.text(4, 3, [" < Hello my child.",
                         " < You're now ten years old.",
                         " < And I think it's now time for you to travel the \
 world and be a Pokete-trainer.",
```
Comment (MaFeLP) [inline pokete.py:953]: Due to adding `mvp.` to line 950, the indentation in these lines changes, which makes them too long:

```suggestion
                            " < You're now ten years old.",
                            " < And I think it's now time for you to travel \
the world and be a Pokete-trainer.",
```

PS: Please also fix indentation of lines 954-968, as GitHub doesn't let me make a suggestion in these lines

File: pokete.py [hunk 8]
```diff
@@ -1341,7 +1000,7 @@ def gen_obs():
                       map_data[ob_map]["soft_obs"][soft_ob])
         for dor in map_data[ob_map]["dors"]:
             parse_obj(_map, dor,
-                      Dor(" ", state="float",
+                      Door(" ", state="float",
                           arg_proto=map_data[ob_map]["dors"][dor]["args"]),
                       map_data[ob_map]["dors"][dor])
```
Comment (MaFeLP) [inline pokete.py:1005]: Also rename this variable from `dor` to `door`

```suggestion
        for door in map_data[ob_map]["dors"]:
            parse_obj(_map, door,
                      Door(" ", state="float",
                           arg_proto=map_data[ob_map]["dors"][door]["args"]),
                      map_data[ob_map]["dors"][door])
```

File: pokete.py [hunk 9]
```diff
@@ -1436,23 +1095,23 @@ def map_additions():
...
     _map.shopdor.add(_map, 61, 6)
 
     # playmap_4
-    _map = ob_maps["playmap_4"]
-    _map.dor_playmap_5 = ChanceDor("~", state="float",
+    _map = obmp.ob_maps["playmap_4"]
+    _map.dor_playmap_5 = ChanceDoor("~", state="float",
                                    arg_proto={"chance": 6,
                                               "map": "playmap_5",
                                               "x": 17, "y": 16})
```
Comment (MaFeLP) [inline pokete.py:1117]: Due to renaming `ChangeDor` to `ChangeDoor`, these lines require one further space of indentation:

```suggestion
                                    arg_proto={"chance": 6,
                                               "map": "playmap_5",
                                               "x": 17, "y": 16})
```

File: pokete.py [hunk 10]
```diff
@@ -1595,18 +1254,18 @@ def map_additions():
...
     _map.inner.add(_map, 0, 0)
 
     # playmap_21
-    _map = ob_maps["playmap_21"]
-    _map.dor_playmap_19 = Dor("_", state="float",
+    _map = obmp.ob_maps["playmap_21"]
+    _map.dor_playmap_19 = Door("_", state="float",
                               arg_proto={"map": "playmap_19",
                                          "x": 26, "y": 1})
-    _map.dor = DorToCenter()
```
Comment (MaFeLP) [inline pokete.py:1608]: Due to changing `Dor` to `Door`, these lines require one further space of indentation:

```suggestion:
                               arg_proto={"map": "playmap_19",
                                          "x": 26, "y": 1})
```

**Comment by MaFeLP** [comment]
I also found, that `pokete_classes.generic_map_handler` is commented in `pokete.py`. Do I have to specifically enable the map handler for it to be tested/used?

**Comment by lxgr-linux** [comment]
Yes, that's true, since it does not have a real usecase as of now it's disabled.

**Review by MaFeLP — CHANGES_REQUESTED** [review]
While play testing for a while, I did not encounter any bugs/problems/crashes.

Maybe add these two doc strings and resolve the conversations afterwards:

- [number one](https://github.com/lxgr-linux/pokete/pull/111/files#r842893561)
- [bumber two](https://github.com/lxgr-linux/pokete/pull/111/files#r842893981)

Also there are some other minor things missing, but it looks very good otherwise!

**Pull request merged**